### PR TITLE
Feature/meta methods

### LINKED ISSUE #14: support metamethods
teal has no way to define supported metamethods yet but https://github.com/teal-language/tl/pull/299 seems to change that so tealr should as well.

### PR DESCRIPTION
**Pull request by lenscas**
fix #14

### CODE REVIEW

File: tealr/src/type_walker.rs [hunk 1]
```diff
@@ -7,30 +7,62 @@ use crate::{
     Direction, TealMultiValue, TealType, TypeBody, TypeName,
 };
 
+pub(crate) fn get_meta_name(name: rlua::MetaMethod) -> &'static str {
+    match name {
+        MetaMethod::Add => "__add",
+        MetaMethod::Sub => "__su",
```
Comment (makspll) [inline tealr/src/type_walker.rs:13]: There is a typo here, this should be "__sub"!

Comment (lenscas) [reply tealr/src/type_walker.rs:13]: that I missed that is... not good.

Good catch. Will try and fix it today :)

Comment (makspll) [reply tealr/src/type_walker.rs:13]: No worries, just found out when generating docs, it doesn't actually mess with documentation just the .d.tl files

Comment (lenscas) [reply tealr/src/type_walker.rs:13]: fixed by #51

**Pull request merged**